vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,136 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <benchmark/benchmark.h>
+#include <spdlog/spdlog.h>
+#include <opencv2/opencv.hpp>
+#include <image_event_stamped.h>
+#include <klepsydra/mem_core/basic_middleware_provider.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
+#include <klepsydra/core/event_transform_forwarder.h>
+
+class FactoryPerformanceTest : public benchmark::Fixture {
+    public:
+        FactoryPerformanceTest() :
+         benchmark::Fixture(),
+         width(1000), 
+         height(500),
+         imageDataFactory(width, height,  cv::IMREAD_REDUCED_GRAYSCALE_2, "raw_image"){
+
+        }
+        ~FactoryPerformanceTest() {}
+        
+        void SetUp(const ::benchmark::State& state) {
+            bool withFactory = state.range(0);
+            if (withFactory) {
+                mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageData>(nullptr,
+                "img", 4, 6, imageDataFactory.initializerFunction, imageDataFactory.eventClonerFunction, false));
+            } else {
+                mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageData>(nullptr,
+                "img", 4, 6, nullptr, nullptr, false));
+            }
+            mp->start();
+            subscriber = mp->getSubscriber();
+        }
+
+        void TearDown(const ::benchmark::State& state) {
+            
+            mp->stop();
+        }
+
+        void setListener(benchmark::State &state) {
+            subscriber->registerListener("receiveIm", [&state, this](const kpsr::vision_ocv::ImageData& im) {
+                    auto now = std::chrono::duration_cast<std::chrono::milliseconds>(
+                    std::chrono::system_clock::now().time_since_epoch()).count();
+                    state.SetIterationTime(now - im.timestamp);
+            });
+        }
+        int width, height;
+        kpsr::vision_ocv::ImageDataFactory imageDataFactory;
+        std::unique_ptr<kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageData>> mp;
+        kpsr::Subscriber<kpsr::vision_ocv::ImageData> * subscriber;
+};
+
+BENCHMARK_DEFINE_F(FactoryPerformanceTest, SimpleTest) (benchmark::State &state)
+{
+    setListener(state);
+    auto publisher = mp->getPublisher();
+    
+    auto &publisherStats = publisher->_publicationStats;
+    auto listenerStats = subscriber->getSubscriptionStats("receiveIm");
+    
+    kpsr::vision_ocv::ImageData event;
+    imageDataFactory.initializerFunction(event);
+    for (auto _ : state) {
+        event.timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::system_clock::now().time_since_epoch()).count();
+        publisher->publish(event);
+        while (listenerStats->totalProcessed < publisherStats.totalProcessed) {
```

Comment:
If we use `EventEmitterMiddlewareProvider`, this is not needed. However, we should also add test for the `EventLoopMiddlewareProvider`, in which case this is definitely necessary.